### PR TITLE
feat(shareLatest): removing distinctShareReplay in favor of shareLatest

### PR DESCRIPTION
**Pull request by josepot**
I'm so so so so so happy about these changes that I can't wait to get them merged! :heart_eyes:

These refactor addresses [what we've discussed](https://github.com/re-rxjs/re-rxjs/issues/41#issuecomment-650778069) on #41 . I think that we finally managed to get this right. Thanks a lot @voliva for the feedback and the help!

FTR: our tests are pretty bad for documenting the behavior of the API and that's something that we should improve ASAP. However, they are great for addressing refactors like this one.

### CODE REVIEW

**Comment by josepot** [comment]
And on top of that: have a look at the new bundle-size :heart_eyes: !!! Yes! this is it! This should be the final API for `v0.2.0`, yes! finally!

File: src/connectObservable.ts [hunk 1]
```diff
@@ -13,30 +12,19 @@ import { ConnectorOptions, defaultConnectorOptions } from "./options"
...
 ) {
-  const _options = {
-    ...defaultConnectorOptions,
-    ...options,
-  }
-  const sharedObservable$ = distinctShareReplay(_options.compare)(
-    concat(observable, NEVER),
-  )
-
+  const sharedObservable$ = shareLatest<T>(false)(concat(observable, NEVER))
```
Comment (voliva) [inline src/connectObservable.ts:23]: with `shareLatest(false)` concat NEVER is redundant, right?

File: test/operators/share-latest.test.ts [hunk 2]
```diff
@@ -11,32 +10,6 @@ const scheduler = () =>
   })
 
 describe("operators/distinctShareReplay", () => {
```
Comment (voliva) [inline test/operators/share-latest.test.ts:12]:
```suggestion
describe("operators/shareLatest", () => {
```

File: src/connectObservable.ts (hunk 1 above)
Comment (josepot) [reply src/connectObservable.ts:23]: yes, indeed! thanks!

**Pull request merged**